core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/dao/OrderItemDaoImpl.java

```diff
@@ -144,4 +144,27 @@ public OrderItemPriceDetail initializeOrderItemPriceDetails(OrderItem item) {
 
         return query.getResultList();
     }
+
+    @Override
+    public Long readNumberOfOrderItems() {
+        CriteriaBuilder builder = em.getCriteriaBuilder();
+        CriteriaQuery<Long> criteria = builder.createQuery(Long.class);
+        criteria.select(builder.count(criteria.from(OrderItemImpl.class)));
+        TypedQuery<Long> query = em.createQuery(criteria);
+        return query.getSingleResult();    }
+
+    @Override
+    public List<OrderItem> readBatchOrderItems(int start, int count) {
```

Comment:
for the purposes of an export - seems like this method also needs to be qualified as well. i.e. if you only export submitted orders, you would likely only want to export order items that belong to those submitted orders as well (not all order items). This should consider order statuses like the order method

Reply:
Added qualification